viewer/src/main/java/nl/b3p/viewer/util/IPAuthenticationFilter.java

```diff
@@ -0,0 +1,347 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package nl.b3p.viewer.util;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpSession;
+import nl.b3p.viewer.config.security.User;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Hibernate;
+import org.stripesstuff.stripersist.Stripersist;
+
+/**
+ *
+ * @author Meine Toonen meinetoonen@b3partners.nl
+ */
+public class IPAuthenticationFilter implements Filter {
+    
+    private static final Log log = LogFactory.getLog(IPAuthenticationFilter.class);
+
+    // The filter configuration object we are associated with.  If
+    // this value is null, this filter instance is not currently
+    // configured. 
+    private FilterConfig filterConfig = null;
+    
+    private static final String IP_CHECK = IPAuthenticationFilter.class + "_IP_CHECK";
+    private static final String USER_CHECK = IPAuthenticationFilter.class + "_USER_CHECK";
+    
+    
+    public IPAuthenticationFilter() {
+    }    
+    
+    /**
+     *
+     * @param r The servlet request we are processing
+     * @param response The servlet response we are creating
+     * @param chain The filter chain we are processing
+     *
+     * @exception IOException if an input/output error occurs
+     * @exception ServletException if a servlet error occurs
+     */
+    public void doFilter(ServletRequest r, ServletResponse response,
+            FilterChain chain)
+            throws IOException, ServletException {
+        
+        log.debug("IPAuthenticationFilter:doFilter()");
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        User u = null;
+        if(session.getAttribute(IP_CHECK) == null  && session.getAttribute(USER_CHECK) == null){
+            
+            String ipAddress = request.getHeader("X-FORWARDED-FOR");
```

Comment:
also, apparently, the forwarded-for can hold multiple adresses [1] so you may want to get the substring up to the first comma.


[1] https://nl.wikipedia.org/wiki/X-Forwarded-For